src/main/java/seedu/trippie/command/AddExpenseCommand.java

```diff
@@ -5,39 +5,58 @@
 import seedu.trippie.data.ExpenseList;
 import seedu.trippie.data.Trip;
 import seedu.trippie.data.TripList;
+import seedu.trippie.exception.TrippieInvalidArgumentException;
 
+import java.util.Collections;
 import java.util.List;
 
 
 public class AddExpenseCommand extends Command {
 
+    private static final String FORMAT_ERROR_MESSAGE = "Incorrect format for [buy] command! Please try the following:\n"
+            + "Format: buy /i ITEM_NAME /c FINAL_COST /d DAY_NUMBER\n"
+            + "Example: buy /i R&B Brown Sugar /c 3.00 /d 2";
+    private static final String PARAMETER_ERROR_MESSAGE = "Please check that your FINAL_COST and DAY_NUMBER parameters"
+            + "are in \nnumerical form.";
+
     private final String expenseName;
-    private final String expenseCost;
-    private final String expenseDayBought;
+    private final Float expenseCost;
+    private final int expenseDayBought;
 
-    public AddExpenseCommand(String userInput) {
-        this.expenseName = extractExpenseName(userInput);
-        this.expenseCost = extractExpenseCost(userInput);
-        this.expenseDayBought = extractDayBought(userInput);
+    public AddExpenseCommand(String userInput) throws TrippieInvalidArgumentException {
+        try {
+            this.expenseName = extractExpenseName(userInput);
+            this.expenseCost = extractExpenseCost(userInput);
+            this.expenseDayBought = extractDayBought(userInput);
+            char[] characters = userInput.toCharArray();
+            if (characters[3] != ' ') {
+                throw new TrippieInvalidArgumentException(FORMAT_ERROR_MESSAGE);
+            }
+        } catch (IndexOutOfBoundsException e) {
+            throw new TrippieInvalidArgumentException(FORMAT_ERROR_MESSAGE);
+        } catch (NumberFormatException e) {
+            throw new TrippieInvalidArgumentException(PARAMETER_ERROR_MESSAGE);
+        }
     }
 
     public String extractExpenseName(String userInput) {
         String withoutCost = userInput.split(" /c ")[0];
         return withoutCost.split(" /i ")[1];
     }
 
-    public String extractExpenseCost(String userInput) {
+    public Float extractExpenseCost(String userInput) {
         String withoutDay = userInput.split(" /d ")[0];
         String expenseCost = withoutDay.split(" /c ")[1];
         if (expenseCost.contains("$")) {
             expenseCost = expenseCost.replace("$", "");
         }
-        return expenseCost;
+        return Float.parseFloat(expenseCost);
+
     }
 
-    public String extractDayBought(String userInput) {
+    public int extractDayBought(String userInput) {
         String onlyDay = userInput.split(" /d ")[1];
-        return onlyDay.replaceAll("[^0-9]","").trim();
```

Comment:
I did this so it gives users the flexibility or options to type in "2" or "Day 2". so I dun think we should replace it?

Reply:
shawn I think there are some of your changes I accidentally deleted. It's the automerge..

Reply:
No, this was changed in this commit. But we could think of this feature again. [Link to commit.](https://github.com/AY2021S1-CS2113T-W11-2/tp/commit/a604d8334a4b5a00dc9181c49b5a63de6bee7a72)